cmd/agent/main.go

```diff
@@ -63,6 +66,16 @@ func main() {
 
 		initFeatures()
 
+		restConfig, err := clientcmd.BuildConfigFromFlags("", os.Getenv("KUBECONFIG"))
```

Comment:
This isn't a reliable way to get a rest config, the agent is running in a pod which uses a mounted service account token, not a kubeconfig.

There is a running rancher instance in the agent which has its own wrangler context, is there some way that could be used?

Reply:
The in-cluster config would work